lib/bdev/nvme/blockdev_nvme.c

```diff
@@ -390,6 +390,10 @@ probe_cb(void *cb_ctx, const struct spdk_nvme_probe_info *probe_info,
 		return false;
 	}
 
+	if (ctx->controllers_remaining > 0) {
```

Comment:
I don't think the controllers_remaining change is necessary, and I think actually the controllers_remaining logic can be deleted entirely - can you skip this part of the patch and verify that it still works correctly?